src/LiveComponent/src/Twig/LiveComponentRuntime.php

```diff
@@ -24,51 +25,51 @@
  */
 final class LiveComponentRuntime
 {
-    private LiveComponentHydrator $hydrator;
-    private ComponentFactory $factory;
-    private UrlGeneratorInterface $urlGenerator;
-    private ?CsrfTokenManagerInterface $csrfTokenManager;
-
-    public function __construct(LiveComponentHydrator $hydrator, ComponentFactory $factory, UrlGeneratorInterface $urlGenerator, CsrfTokenManagerInterface $csrfTokenManager = null)
-    {
-        $this->hydrator = $hydrator;
-        $this->factory = $factory;
-        $this->urlGenerator = $urlGenerator;
-        $this->csrfTokenManager = $csrfTokenManager;
+    public function __construct(
+        private Environment $twig,
+        private LiveComponentHydrator $hydrator,
+        private ComponentFactory $factory,
+        private UrlGeneratorInterface $urlGenerator,
+        private ?CsrfTokenManagerInterface $csrfTokenManager = null
+    ) {
     }
 
-    public function renderLiveAttributes(Environment $env, object $component, string $name = null): string
+    public function renderLiveAttributes(array $context): string
     {
-        $name = $this->nameFor($component, $name);
-        $url = $this->urlGenerator->generate('live_component', ['component' => $name]);
-        $data = $this->hydrator->dehydrate($component);
-
-        $ret = sprintf(
-            'data-controller="live" data-live-url-value="%s" data-live-data-value="%s"',
-            twig_escape_filter($env, $url, 'html_attr'),
-            twig_escape_filter($env, json_encode($data, \JSON_THROW_ON_ERROR), 'html_attr'),
-        );
-
-        if (!$this->csrfTokenManager) {
-            return $ret;
+        if (!isset($context['_component_config'])) {
+            throw new \LogicException('init_live_component can only be called within a component template.');
         }
 
-        return sprintf('%s data-live-csrf-value="%s"',
-            $ret,
-            $this->csrfTokenManager->getToken($name)->getValue()
-        );
+        return $this->getLiveAttributes($context['this'], $context['_component_config']);
     }
 
     public function getComponentUrl(object $component, string $name = null): string
     {
         $data = $this->hydrator->dehydrate($component);
-        $params = ['component' => $this->nameFor($component, $name)] + $data;
+        $params = ['component' => $this->factory->configFor($name ?? $component)] + $data;
 
         return $this->urlGenerator->generate('live_component', $params);
     }
 
-    private function nameFor(object $component, string $name = null): string
+    public function getLiveAttributes(object $component, array $config): ComponentAttributes
     {
-        return $this->factory->configFor($component, $name)['name'];
+        if (!isset($config['live'])) {
+            throw new \LogicException(sprintf('"%s" is not a Live Component.', $config['class']));
+        }
+
+        $url = $this->urlGenerator->generate('live_component', ['component' => $config['name']]);
```

Comment:
It seems like this and `getComponentUrl()` should share some code, right? Is that method still kind of a TODO to fix?

Reply:
Yes, that twig function needs a big overhaul but wanted to save for another PR.